scripts/vipers/vipers.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# TODO: Address all TODOs and remove all explanatory comments
+"""TODO: Add a description here."""
+
+
+import csv
+import json
+import os
+import datasets
+from datasets.data_files import DataFilesPatternsDict
+from datasets import Features, Value, Array2D, Sequence
+import itertools
+import h5py
+import numpy as np
+
+# TODO: Add BibTeX citation
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+@InProceedings{huggingface:dataset,
+title = {A great new dataset},
+author={huggingface, Inc.
+},
+year={2020}
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+The VIPERS spectra catalog
+"""
+
+# TODO: Add a link to an official homepage for the dataset here
+_HOMEPAGE = ""
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_VERSION = "0.0.1"
+
+_FLOAT_FEATURES = [
+    'ra',
+    'dec',
+    'redshift', 
+    'redflag', 
+    'exptime', 
+    'norm', 
+    'mag'
+]
+
+class VIPERS(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="vipers_w1",
+                               version=VERSION,
+                               data_files=DataFilesPatternsDict.from_patterns({'train': ['vipers_w1/healpix=*/*.h5']}),
+                               description="VIPERS W1 Catalog"),
+        datasets.BuilderConfig(name="vipers_w4",
+                               version=VERSION,
+                               data_files=DataFilesPatternsDict.from_patterns({'train': ['vipers_w4/healpix=*/*.h5']}),
+                               description="VIPERS W4 Catalog"),
+        datasets.BuilderConfig(name="all",
+                               version=VERSION,
+                               data_files=DataFilesPatternsDict.from_patterns({'train': ['*/healpix=*/*.h5']}),
+                               description="VIPERS Full Catalog")
+    ]
+
+    DEFAULT_CONFIG_NAME = "all"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
+    def _info(self):
+        """Defines the dataset info."""
+        features = datasets.Features(
+            {
+                "spectrum": Sequence({
```

Comment:
@lhparker1 it looks good, can you just use this formatting of the data to avoid Sequences:
https://github.com/AstroPile/AstroPile_prototype/blob/61e9c9a2928ebf5dd3b3778ccf50345e8b0ea852/scripts/sdss/sdss.py#L132